.eslintrc.yaml

```diff
@@ -106,7 +106,7 @@ rules:
 
   # React Native.  This plugin isn't included in the airbnb config, and
   # nothing is enabled by default, so we enable its rules.
-  react-native/no-inline-styles: error
+  react-native/no-inline-styles: off # Noisy and a bad guide to rerendering optimizations.
```

Comment:
e2dc155335100efca9c8256c07e0f03f3520c1b3

I realized this morning that this may be more detail than is currently, and may be distracting. The `style` prop is only one prop, and there are plenty of other props that can be guilty of causing unnecessary rerenders, and this rule has never caught those.

One argument for keeping all this detail (to the extent that it's correct and readable) is that we may use other eslint rules, like https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md, that handle those other props, narrowing things down to a focus on the style prop.

Reply:
I think a good medium for the material that makes up most of this commit message would be a chat thread, on perhaps `#mobile-team`. That's a good place for putting a lot of detail, and for people to reply and discuss; and Zulip's search makes it pretty accessible later, much more accessible than e.g. a thread on GitHub.

Then for the commit message, I think a summary at the scale of a couple of paragraphs will be good. I think the key points are basically just:
* Some styles are most readable when inlined.
* You might think this is a performance issue, because we're making new style objects on each render. But the reality is we have tons of code that does that already, because the lint rule doesn't catch even simple variations. Moreover a lot of them are for fundamental reasons, like that the style depends on props, state, and context -- so there's not a clear path to turning this rule into something useful.

About the details: does it really matter whether the child is a `PureComponent`? I thought the `PureComponent` optimization was only a shallow check of `props` itself, so that it checks identity on each individual prop value. But perhaps that chat thread will be the best place to discuss :slightly_smiling_face: